### PR TITLE
[COST-7684] Validate S3 credentials, ListBuckets, and JWKS probes

### PR DESCRIPTION
**Pull request by ELK4N4**
## Summary
Closes the ticket-scoped COST-7684 gaps ([gap analysis](https://github.com/project-koku/koku-service-operator/blob/cost-7684-s3-listbuckets-probe/docs/gap_analysis/COST-7684.md)):

- **G1** — AWS SDK `ListBuckets` against the resolved endpoint (user-provided or Discovery OBC/NooBaa) so `StorageReady` is `StorageReachable` / `StorageUnreachable`, not “keys exist”.
- **G2** — Validate object-storage Secrets for `access-key` / `secret-key` before any network call (`StorageSecretInvalid`).
- **G3** — OIDC probe uses the reconcile context, requires HTTP 2xx, and accepts the body only if it is JWKS JSON with a non-empty `keys` array.

S3 / OIDC failures stay non-blocking (same as Kafka). Path-style addressing is used for MinIO / NooBaa / Ceph RGW.

Related, **not** in this PR: R1 (Edge overwrites `AuthenticationReady`), R2 (Kafka multi-broker host parsing), R3 (conditional ROS/Kruize DB keys → COST-8054).

## Test plan
- [x] `go test ./internal/controller/ -count=1 -timeout 60s -run 'JWKS|OIDC|S3|Validation'`
- [ ] On BYOI MinIO: `StorageReady=True` reason `StorageReachable`
- [ ] Missing `secret-key`: `StorageSecretInvalid` (no ListBuckets)
- [ ] Wrong endpoint / MinIO down: `StorageUnreachable`, pipeline still proceeds
- [ ] NooBaa HTTPS: confirm operator pod trusts the OpenShift service CA (otherwise `x509` → `StorageUnreachable`)
- [ ] Keycloak JWKS 200 with empty/`{}` keys → `AuthenticationReady=False` / `OIDCUnreachable`

## Summary by Sourcery

Add non-blocking S3 ListBuckets connectivity validation and stronger JWKS-based OIDC probing to reconcile validation, including credential checks and updated gap-analysis docs.

New Features:
- Introduce S3 connectivity probing using AWS SDK ListBuckets against the resolved endpoint to drive StorageReady conditions.
- Add JWKS-aware OIDC probe that validates HTTP 2xx responses and non-empty JWKS keys before marking AuthenticationReady.

Enhancements:
- Extend validation to verify object storage Secrets contain required access-key and secret-key before attempting S3 calls.
- Update validation to use reconcile context for OIDC JWKS probes and enforce response size and JSON structure constraints.
- Refine validation gating so Kafka, OIDC, and S3 failures remain non-blocking while still updating per-dependency conditions.

Build:
- Add AWS SDK for Go v2 S3-related modules to support ListBuckets-based validation.

CI:
- Update govulncheck workflow configuration to account for a newly ignored vulnerability ID.

Documentation:
- Refresh COST-7684 gap analysis to mark ticket-scoped gaps as closed, documenting the new S3 and OIDC validation behavior and remaining related items.

Tests:
- Add comprehensive tests for JWKS probe behavior and OIDC condition handling, including invalid JSON, empty keys, and cancelled contexts.
- Add S3 validation and ListBuckets probe tests covering reachable, forbidden, unreachable, TLS failures, and discovered-secret scenarios.

### CODE REVIEW

File: internal/controller/validation_s3_test.go [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+)
+
+func s3CredsSecret(name string) *corev1.Secret {
+	return &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{Name: name, Namespace: testNamespace},
+		Data: map[string][]byte{
+			"access-key": []byte(s3TestAccessKey),
+			"secret-key": []byte(s3TestSecretKey),
+		},
+	}
```
[the remark this answers is not in the document]
Comment (ELK4N4) [reply internal/controller/validation_s3_test.go:38]: This is a false positive. `bundledNoKafkaSpec` is defined once in this file (`validation_s3_test.go`). `validation_test.go` only calls it.

Both files are `package controller` `_test.go`, so they already share that helper. CI Test passed; there is no duplicate symbol.

**Comment by martinpovolny** [comment]
Thank you for the PR!
 
ping @dchorvat1 : Daniel, since you aready did some testing of the S3 detection cascade, bringing this to your attention.

**Review by martinpovolny — APPROVED** [review]
